src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -127,6 +127,19 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
     if (!layerParams) {
       throw new Error('Layer params could not be found');
     }
+
+    const dataProductUrl = layerParams['dataProduct'];
+
+    if (!this.evalscript && dataProductUrl) {
+      const response = await fetchDataProduct(dataProductUrl, reqConfig);
+      const evalScript = response?.data?.evalScript;
+      if (evalScript) {
+        this.evalscript = evalScript;
```

Comment:
Would it make sense to set evalscript to `layerParams` object that is then returned and not to the object?

Every piece of code where this function is called, then deals with setting the evalscript to the object if it's not set yet.

True, with this approach, setting `dataProduct` to `null` would need to be added to more places.

But I think it's worth it - to not pollute the function that is responsible only for fetching the data with setting object properties (following separation of concerns).

Reply:
Maybe I could unwind this logic a bit then